### ISSUE TITLE
AnimateDiff/Batch won't work with Controlnet SEGS and Mask to SEGS (Combined)

### ISSUE THREAD

**Issue by oliverban** [ltdrdata/ComfyUI-Impact-Pack#736]
Hello!

First off, thanks so much for a wonderful extension pack!

I am trying to get this head mask I made in 3D to become a SEGS for a "Detailer for AnimateDiff" but it keeps failing to see the segs. First off it takes an enormous amount of time to even go through the "Controlnet Apply (SEGS)" and once it does the image preview on the Detailer is all wrong, showing a different all skewed part of the frame.

Attached in a screenshot. The only thing coming in on the LEFT is the original video that I want to fix the face on. The video coming in at the top is just a green mask of the head going to a Mask from Color and then into a Combined MASK TO SEGS.

I KNOW this should work and that it has something to do with LISTS and BATCHES but I am at my wits end! Thanks for all the help!

FYI: I tested this exact setup (but normal Detailer, not Animatediff Version) and it worked perfectly! :) I just want it to work with AnimateDiff/Batches as well!

<img width="537" alt="brave_JmRwWVR71h" src="https://github.com/user-attachments/assets/1fc1819e-aa9e-4edc-9e5d-a8974f15c312">

**Comment by oliverban** [comment]
Bump? :) Or is it not doable?

**Comment by ltdrdata** [comment]
Have you been read this description?

https://github.com/ltdrdata/ComfyUI-extension-tutorials/blob/Main/ComfyUI-Impact-Pack/tutorial/detectors.md#simple-detector-for-animatediff

**Comment by oliverban** [comment]
> Have you been read this description?
> 
> https://github.com/ltdrdata/ComfyUI-extension-tutorials/blob/Main/ComfyUI-Impact-Pack/tutorial/detectors.md#simple-detector-for-animatediff

Yes I have! But it doesn't say anything on using your own masks, only describes how to use the detector one with BBOX? I want to use my own masks instead since the Simple Detector also fails on detecting a lot of the frames. I was hoping to be able to use the Mask to SEGS one.

**Comment by ltdrdata** [comment]
![image](https://github.com/user-attachments/assets/9d1833cf-bf0b-41d8-bfe6-b12cadead85a)
Do you know this node?

The AnimateDiff version differs in whether it handles the mask as a batch or not. The mask contained in a single SEGS becomes a mask with the same batch size as the input batch image.

**Comment by oliverban** [comment]
> ![image](https://private-user-images.githubusercontent.com/128333288/368808300-9d1833cf-bf0b-41d8-bfe6-b12cadead85a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNDUyNzEsIm5iZiI6MTcyNzA0NDk3MSwicGF0aCI6Ii8xMjgzMzMyODgvMzY4ODA4MzAwLTlkMTgzM2NmLWJmMGItNDFkOC1iZmU2LWIxMmNhZGVhZDg1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyMlQyMjQyNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTYxYjlkYTVhYTI1YTU2MDBiYjNhN2U2YTdjZDEyZWU4N2VmYzk0MTA5MjcwN2RmNjFjNWUzZmFiODk4ZWJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ncADXPHvT2idQCPy77SWxJHzlq4T43Oo43aMo69XLH8) Do you know this node?
> 
> The AnimateDiff version differs in whether it handles the mask as a batch or not. The mask contained in a single SEGS becomes a mask with the same batch size as the input batch image.

Hmm, haven't seen the one with AnimateDiff attached. I will try it out. Thanks! <3
